<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1653</width>
    <height>1162</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="6" column="3">
     <widget class="QPushButton" name="pushButton">
      <property name="text">
       <string>Сгенерировать дерево</string>
      </property>
     </widget>
    </item>
    <item row="0" column="3">
     <widget class="QGroupBox" name="groupBox_2">
      <property name="title">
       <string>Готовые деревья</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_3">
       <item row="0" column="0">
        <widget class="QComboBox" name="trees_combo_box"/>
       </item>
      </layout>
     </widget>
    </item>
    <item row="5" column="3">
     <widget class="QGroupBox" name="groupBox_4">
      <property name="title">
       <string>Геометрия дерева</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_4">
       <item row="0" column="1">
        <widget class="QDoubleSpinBox" name="base_radius_spin_box"/>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_7">
         <property name="text">
          <string>Коэффициент уменьшения</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QDoubleSpinBox" name="min_leaf_radius_spin_box"/>
       </item>
       <item row="1" column="1">
        <widget class="QDoubleSpinBox" name="radius_decay_spin_box"/>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="label_6">
         <property name="text">
          <string>Базовый радиус</string>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QDoubleSpinBox" name="gravity_factor_spin_box"/>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_8">
         <property name="text">
          <string>Минимальный радиус</string>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="label_9">
         <property name="text">
          <string>Фактор гравитации</string>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QLabel" name="label_10">
         <property name="text">
          <string>Сегменты окружности</string>
         </property>
        </widget>
       </item>
       <item row="4" column="1">
        <widget class="QDoubleSpinBox" name="radial_segments_spin_box"/>
       </item>
      </layout>
     </widget>
    </item>
    <item row="1" column="3">
     <widget class="QGroupBox" name="groupBox_5">
      <property name="title">
       <string>Параметры освещения</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_5">
       <item row="1" column="0" colspan="3">
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QLabel" name="label_12">
           <property name="text">
            <string>x</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_13">
           <property name="text">
            <string>y</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_14">
           <property name="text">
            <string>z</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="0" column="1" colspan="2">
        <widget class="QDoubleSpinBox" name="intensity_spin_box"/>
       </item>
       <item row="2" column="0" colspan="3">
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QDoubleSpinBox" name="x_light_spin_box"/>
         </item>
         <item>
          <widget class="QDoubleSpinBox" name="y_light_spin_box"/>
         </item>
         <item>
          <widget class="QDoubleSpinBox" name="z_light_spin_box"/>
         </item>
        </layout>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="label_11">
         <property name="text">
          <string>Интенсивность:</string>
         </property>
        </widget>
       </item>
       <item row="3" column="0" colspan="3">
        <widget class="QPushButton" name="light_push_button">
         <property name="text">
          <string>Применить</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="3" column="3">
     <widget class="QGroupBox" name="groupBox_3">
      <property name="title">
       <string>Камеры</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QRadioButton" name="orbit_camera_radio_btn">
         <property name="text">
          <string>Орбитальная камера</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="free_camera_radio_btn">
         <property name="text">
          <string>Свободная камера</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="4" column="3">
     <widget class="QGroupBox" name="groupBox">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="title">
       <string>Параметры L-системы</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="5" column="2">
        <widget class="QDoubleSpinBox" name="angle_spin_box"/>
       </item>
       <item row="3" column="0" colspan="3">
        <widget class="QPushButton" name="add_rule_btn">
         <property name="text">
          <string>Добавить правило</string>
         </property>
        </widget>
       </item>
       <item row="6" column="2">
        <widget class="QDoubleSpinBox" name="step_spin_box"/>
       </item>
       <item row="2" column="0" colspan="3">
        <widget class="QTableView" name="rules_table"/>
       </item>
       <item row="1" column="2">
        <widget class="QLineEdit" name="rule_edit"/>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>Аксиома:</string>
         </property>
        </widget>
       </item>
       <item row="5" column="0">
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>Угол:</string>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Кол-во итераций:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_5">
         <property name="text">
          <string>Правило</string>
         </property>
        </widget>
       </item>
       <item row="4" column="2">
        <widget class="QDoubleSpinBox" name="iterations_spin_box"/>
       </item>
       <item row="6" column="0">
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>Длина шага:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QComboBox" name="symbols_combo_box"/>
       </item>
       <item row="0" column="1" colspan="2">
        <widget class="QLineEdit" name="axiome_edit"/>
       </item>
      </layout>
     </widget>
    </item>
    <item row="0" column="4" rowspan="7">
     <widget class="QGraphicsView" name="graphicsView">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>1</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="verticalScrollBarPolicy">
       <enum>Qt::ScrollBarPolicy::ScrollBarAlwaysOff</enum>
      </property>
      <property name="horizontalScrollBarPolicy">
       <enum>Qt::ScrollBarPolicy::ScrollBarAlwaysOff</enum>
      </property>
      <property name="backgroundBrush">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>255</red>
         <green>255</green>
         <blue>255</blue>
        </color>
       </brush>
      </property>
     </widget>
    </item>
    <item row="2" column="3">
     <widget class="QGroupBox" name="groupBox_6">
      <property name="title">
       <string>Тени</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_6">
       <item row="0" column="0">
        <widget class="QCheckBox" name="enable_shadows_check_box">
         <property name="text">
          <string>Наличие теней</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1653</width>
     <height>26</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
